packages/react-compose/src/next/getSlots.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { getNativeElementProps } from '@uifabric/utilities';
+import { GenericDictionary } from './types';
+import { nullRender } from './nullRender';
+
+/**
+ * Given the state and an array of slot names, will break out `slots` and `slotProps`
+ * collections. Ensures that each slot is a renderable component, and each slotProp is
+ * a list of acceptable native properties for the given slot. Native properties are only
+ * filtered if the slot is rendered "as" a string type, and will use `getNativeElementProps`
```

Comment:
`Native properties are only filtered if the slot is rendered "as" a string type`

You're doing this check for the slot `as` prop but not for the `root`.

Reply:
Fixed.